internal/server/server.go

```diff
@@ -24,27 +25,58 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-func registerHandler(ctx context.Context, s *server.MuxServer, gw *server.GRPCGateway, deps api.Deps) {
-	s.RegisterHandler("/admin*", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintf(w, "pong")
-	}))
-
-	s.RegisterHandler("/admin/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintf(w, "pong")
-	}))
+/*
+	func registerHandler(ctx context.Context, s *server.MuxServer, gw *server.GRPCGateway, deps api.Deps) {
+		s.RegisterHandler("/admin*", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintf(w, "pong")
+		}))
 
-	// grpc gateway api will have version endpoints
-	s.SetGateway("/admin", gw)
-	v1beta1.Register(ctx, s, gw, deps)
-}
+		s.RegisterHandler("/admin/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintf(w, "pong")
+		}))
 
+		// grpc gateway api will have version endpoints
+		s.SetGateway("/admin", gw)
+		v1beta1.Register(ctx, s, gw, deps)
+	}
+*/
 func Serve(
 	ctx context.Context,
 	logger log.Logger,
 	cfg Config,
 	nrApp newrelic.Application,
 	deps api.Deps,
-) (*server.MuxServer, error) {
+) error {
+	httpMux := http.NewServeMux()
+	httpMux.Handle("/admin*", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "pong")
+	}))
+
+	httpMux.Handle("/admin/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "pong")
+	}))
+
+	grpcGateway := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(customHeaderMatcherFunc(map[string]bool{cfg.IdentityProxyHeader: true})))
+	httpMux.Handle("/admin/", http.StripPrefix("/admin", grpcGateway))
+	grpcServer := grpc.NewServer(getGRPCMiddleware(cfg, logger, nrApp))
+	address := fmt.Sprintf("%s:%d", cfg.Host, cfg.Port)
+
+	v1beta1.Register(ctx, address, grpcServer, grpcGateway, deps)
+
+	go mux.Serve(
```

Comment:
Do we need to run this as a goroutine?

Reply:
I'll move it out of the go routine, as we discussed.